pinc/unicode.inc

```diff
@@ -170,3 +170,15 @@ function guess_string_encoding($text)
     # Give up and return ISO-8859-1
     return 'ISO-8859-1';
 }
+
+function codepoints_to_hex($string)
+{
+    // find the components of combined character as hex strings
+    $codepoints = [];
+    preg_match_all("/./u", $string, $matches);
+    foreach($matches[0] as $match)
+    {
+        $codepoints[] = str_pad(dechex(IntlChar::ord($match)), 4, "0", STR_PAD_LEFT);
```

Comment:
Please check in `portable-utf8.php` as I suspect there is already a function that does this.

Reply:
There is something useful.